flaps/flaps_machines.go

```diff
@@ -17,9 +19,165 @@ import (
 
 var NonceHeader = "fly-machine-lease-nonce"
 
-func (f *Client) sendRequestMachines(ctx context.Context, method, endpoint string, in, out interface{}, headers map[string][]string) error {
-	endpoint = fmt.Sprintf("/apps/%s/machines%s", f.appName, endpoint)
-	return f._sendRequest(ctx, method, endpoint, in, out, headers)
+type flapsAction int
+
+func (a *flapsAction) String() string {
+	switch *a {
+	case launch:
+		return "launch"
+	case update:
+		return "update"
+	case start:
+		return "start"
+	case wait:
+		return "wait"
+	case stop:
+		return "stop"
+	case restart:
+		return "restart"
+	case get:
+		return "get"
+	case list:
+		return "list"
+	case destroy:
+		return "destroy"
+	case kill:
+		return "kill"
+	case findLease:
+		return "findLease"
+	case acquireLease:
+		return "acquireLease"
+	case refreshLease:
+		return "refreshLease"
+	case releaseLease:
+		return "releaseLease"
+	case exec:
+		return "exec"
+	case ps:
+		return "ps"
+	case cordon:
+		return "cordon"
+	case uncordon:
+		return "uncordon"
+	default:
+		return "unknownAction"
+	}
+}
+
+const (
+	launch flapsAction = iota
+	update
+	start
+	wait
+	stop
+	restart
+	get
+	list
+	destroy
+	kill
+	findLease
+	acquireLease
+	refreshLease
+	releaseLease
+	exec
+	ps
+	cordon
+	uncordon
+)
+
+// The mapping from a flaps action to the API endpoint being hit
+var flapsActionToEndpoint = map[flapsAction]string{
+	launch:       "",
+	update:       "",
+	start:        "start",
+	wait:         "wait",
+	stop:         "stop",
+	restart:      "restart",
+	get:          "",
+	list:         "",
+	destroy:      "",
+	kill:         "signal",
+	findLease:    "lease",
+	acquireLease: "lease",
+	refreshLease: "lease",
+	releaseLease: "lease",
+	exec:         "exec",
+	ps:           "ps",
+	cordon:       "cordon",
+	uncordon:     "uncordon",
+}
+
+// FIXME: rename this, bad name.
+type flapsActionInfo struct {
+	action          flapsAction
+	machineID       string
+	queryParameters map[string]string
+}
```

Comment:
this type is half way to become the param's struct for sendRequestMachines function

It could absorb Method and Headers too, in fact, Method can be derived from Action as it does for endpoint.